ext/thymeleaf/src/main/java/org/eclipse/krazo/ext/thymeleaf/JakartaServletWebApplicationWrapper.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2014-2015 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2018, 2019 Eclipse Krazo committers and contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.krazo.ext.thymeleaf;
+
+import org.thymeleaf.web.servlet.JakartaServletWebApplication;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.http.HttpServletRequest;
+
+/**
+ * Wrapper to circumvent <code>JakartaServletWebApplication</code> not being proxiable by CDI (because of it being final).
+ *
+ * @author Jeremias Weber
+ */
+@ApplicationScoped
+public class JakartaServletWebApplicationWrapper {
+
+    private JakartaServletWebApplication application;
+
+    public JakartaServletWebApplicationWrapper() {}
+
+    @Inject
+    public JakartaServletWebApplicationWrapper(final HttpServletRequest request) {
```

Comment:
Maybe a producer should be used so that people can override it more easily? If so, should it be annotated with `ViewEngineConfig`?

Reply:
@chkal @erdlet should a producer be used and if so, should it be annotated with `ViewEngineConfig`? Maybe I should define an interface for the wrapper so that people can override it?

Reply:
Sorry for the delayed answer. A producer should be enough, as you can overwrite the produced instance with the `@Specialized` annotation.

I hope I'll find some time to have a deeper look into the PR later today.

Reply:
So from my side this looks fine. Only need to check if we have to create a CQ as I asked @chkal below.